sdk/trace/span.go

```diff
@@ -588,20 +593,21 @@ type samplingData struct {
 
 func makeSamplingDecision(data samplingData) SamplingResult {
 	sampler := data.cfg.DefaultSampler
-	spanContext := &data.span.spanContext
-	sampled := sampler.ShouldSample(SamplingParameters{
+	return sampler.ShouldSample(SamplingParameters{
 		ParentContext:   data.parent,
-		TraceID:         spanContext.TraceID,
+		TraceID:         data.span.spanContext.TraceID,
 		Name:            data.name,
 		HasRemoteParent: data.remoteParent,
 		Kind:            data.kind,
 		Attributes:      data.attributes,
 		Links:           data.links,
 	})
-	if sampled.Decision == RecordAndSample {
-		spanContext.TraceFlags |= trace.FlagsSampled
-	} else {
-		spanContext.TraceFlags &^= trace.FlagsSampled
-	}
-	return sampled
+}
+
+func isRecording(s SamplingResult) bool {
+	return s.Decision == RecordOnly || s.Decision == RecordAndSample
+}
+
+func isSampled(s SamplingResult) bool {
+	return s.Decision == RecordAndSample
```

Comment:
Should these be methods on `SamplingResult`?

Reply:
I did have them that way initially, but we don't have any other methods on `SamplingResult` and I think my mind drifted [back to this presentation](https://www.youtube.com/watch?v=WLDT1lDOsb4&ab_channel=CppCon) I saw a few years ago.

I'm happy to change them into methods though if you think it's more Go-idiomatic.

Reply:
I'm not sure whether it's more idiomatic, it just feels to me like these are coupled to the data held in the struct and make sense only in relation to it.  `isRecording(sr)` and `sr.isRecording()` are equivalent, but the latter scans better to me.  If they were exported I would definitely say that the method form is preferred because `trace.IsRecording(sr)` doesn't read right while `sr.IsRecording()` does.

I don't think this is a hill I need to die on, though.  @MrAlias care to weigh in?

Reply:
I don't have a preference, I can see it both ways. I usually default to the author the PR having to some leeway for choosing the design of these kinds of things that are unexported. I think we would need to document the choice if we want to make a recommendation here so people can understand the expectation, and that is getting into the level-of-effort that I don't think is warranted.

Reply:
I agree if it were exported, I would definitely make these methods on `SamplingResult`.

Reply:
@Aneurysm9 okay resolving this?